src/transformers/modeling_reformer.py

```diff
@@ -349,18 +392,73 @@ def forward(
         # num hashes can optionally be overwritten by user
         num_hashes = num_hashes if num_hashes is not None else self.num_hashes
 
-        # project hidden_states to query_key and value
-        query_key_vectors = self.query_key(hidden_states)
-        value_vectors = self.value(hidden_states)
+        do_cached_attention = use_cache and past_buckets_states[1] is not None
+
+        # check if cache shall be used and that hidden states are already cached
+        if do_cached_attention:
```

Comment:
why not always use the cache?

Reply:
Even if `use_cache=True`, the first inference step will have `past_buckets_states == None` => so it should not do `do_cached_attention`.